component/file_cache/file_cache.go

```diff
@@ -891,6 +1026,35 @@ func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Hand
 	return handle, nil
 }
 
+func (fc *FileCache) CreateDir(options internal.CreateDirOptions) error {
+
+	nextAttr := FileAttributes{}
+	nextAttr.operation = "CreateDir"
+	nextAttr.options = options
+
+	_, loaded := fc.fileOps.LoadOrStore(options.Name, nextAttr)
+
+	if loaded {
+
+		fc.fileOps.Delete(options.Name)
+		fc.fileOps.Store(options.Name, nextAttr)
+	}
+	fc.asyncSignal.TryLock() // Make sure we don't unlock a mutex that is not locked
+	fc.asyncSignal.Unlock()  // Signal to async thread to do work
```

Comment:
Yeah it's a bit of a weird workaround to try and stall the async thread from spinning until there's work to be done. Normally a mutex indicates that we want to lock other threads from accessing a critical section, but in this use case we are using it as a signal that we can begin to do work. I used a mutex because they automatically put the thread to sleep and awake when unlocked, so we don't need to do any sleep management on our own. I can go more in depth with the specific implementation

Reply:
Is there a reason you chose to do it this way, rather than with a channel?